work17/index.js

```diff
@@ -0,0 +1,191 @@
+
+const renderLoading = () => {
+    const div = document.createElement("div");
+    const img = document.createElement("img");
+
+    div.classList = "loading-wrap";
+    img.src = "./img/loading-circle.gif";
+
+    wrapDiv.appendChild(div).appendChild(img);
+    document.body.appendChild(div);
+}
+
+const hideLoadingImg = () => {
+    document.querySelector(".loading-wrap").remove();
+}
+
+const init = async () => {
+    renderLoading();
+    let imgJsonData;
+    try {
+        imgJsonData = await getRequestData(REQUEST_URL);
+    } catch (error) {
+        renderErrorMessage(`${error.message}`)
+    } finally {
+        hideLoadingImg();
+    }
+
+    if (!imgJsonData.data.length) {
+        renderErrorMessage("JSONデータが空です。");
+        return;
+    }
+    renderSlideImg(imgJsonData.data);
+}
+
+const getRequestData = async REQUEST_URL => {
+    return new Promise(resolve => {
+        setTimeout(() => {
+            resolve(fetchSlideImgData(REQUEST_URL));
+        }, 3000);
+    })
+}
+
+const fetchSlideImgData = async () => {
+    const parameter = {
+        headers: {
+            'Content-Type': 'application/json'
+        }
+    }
+    const response = await fetch(REQUEST_URL, parameter);
+    if (!response.ok) {
+        renderErrorMessage(`${response.status}:${response.statusText}`);
+    } else {
+        const jsonData = response.json();
+        return jsonData;
+    }
+}
+
+const renderErrorMessage = errorMessage => {
+    const p = document.createElement("p");
+
+    p.textContent = errorMessage;
+    document.body.prepend(p);
+}
+
+const renderSlideImg = imagesDataArray => {
+    const slideDiv = document.createElement("div");
+    const ul = document.createElement("ul");
+    const fragment = document.createDocumentFragment();
+
+    ul.classList = "slider_item"
+    slideDiv.classList = "slider_slide";
+
+    for (let i = 0; i < imagesDataArray.length; i++) {
+        const li = document.createElement("li");
+        const img = document.createElement("img");
+
+        img.src = imagesDataArray[i].src;
+        li.classList = "slider_items";
+        if (!i) {
+            li.classList.add("is-display");
+        }
+        li.id = `js-sliderItems${i + 1}`;
+
+        li.appendChild(img);
+        fragment.appendChild(li);
+    }
+
+    document.body.appendChild(wrapDiv)
+        .appendChild(slideDiv).appendChild(ul).appendChild(fragment);
+    slideDiv.appendChild(renderPagingNumber());
+    renderSlideArrow();
+    clickedSliderEvent();
+}
+
+const renderPagingNumber = () => {
+    let count = 1;
+    const sliderListElements = [...document.querySelectorAll(".slider_items")];
+
+    const span = document.createElement("span");
+    span.classList = "slider_pageNum";
+
+    sliderListElements.forEach(element => {
+        if (element.classList.contains("is-display")) {
+            span.textContent = `${count} / ${sliderListElements.length}`;
+            return span;
+        }
+    });
```

Comment:
Q.
Why is this conditional branching necessary?

Reply:
Since we were using a forEach statement, we thought we needed a conditional branch that would produce a result when there was an "is-display".
However, this statement was pointless.

commit: [cb85381](https://github.com/yasu-pro/js-handson/pull/22/commits/cb85381d701e23c42044bb5d586b05e36f5f6561)